iOS_SDK/OneSignalSDK/Source/OneSignal.m

```diff
@@ -464,90 +473,120 @@ + (BOOL)coldStartFromTapOnNotification {
+ */
++ (void)setAppId:(nonnull NSString*)newAppId {
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"setAppId(id) called with appId: %@!", newAppId]];
+
+    if (!newAppId || newAppId.length == 0) {
+        [OneSignal onesignal_Log:ONE_S_LL_WARN message:@"appId set, but please call setLaunchOptions(launchOptions) to complete OneSignal init!"];
+        return;
+    } else if (appId && ![newAppId isEqualToString:appId])  {
+        // Pre-check on app id to make sure init of SDK is performed properly
+        //     Usually when the app id is changed during runtime so that SDK is reinitialized properly
+        initDone = false;
+    }
+
+    appId = newAppId;
+
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"setAppId(id) finished, checking if launchOptions has been set before proceeding...!"];
+    if (!hasSetLaunchOptions) {
+        [OneSignal onesignal_Log:ONE_S_LL_WARN message:@"appId set, but please call setLaunchOptions(launchOptions) to complete OneSignal init!"];
+        return;
+    }
+
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"setAppId(id) successful and launchOptions are set, continuing OneSignal init..."];
+    [self init];
+}
+
+/*
+ 1/2 steps in OneSignal init, relying on setAppId (usage order does not matter)
+ Sets the iOS sepcific app settings
+ Method must be called to successfully init OneSignal
+ */
++ (void)setLaunchOptions:(nullable NSDictionary*)newLaunchOptions {
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:[NSString stringWithFormat:@"setLaunchOptions() called with launchOptions: %@!", launchOptions.description]];
+
+    launchOptions = newLaunchOptions;
+    hasSetLaunchOptions = true;
+    
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"setAppId(id) finished, checking if launchOptions has been set before proceeding...!"];
+    if (!appId || appId.length == 0) {
+        // Read from .plist if not passed in with this method call
+        appId = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"OneSignal_APPID"];
+        if (!appId) {
+
+            appId = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"GameThrive_APPID"];
+            if (!appId) {
+
+                let prevAppId = [OneSignalUserDefaults.initStandard getSavedStringForKey:OSUD_APP_ID defaultValue:nil];
+                if (!prevAppId) {
+                    [OneSignal onesignal_Log:ONE_S_LL_WARN message:@"launchOptions set, but please call setAppId(appId) with a valid appId to complete OneSignal init!"];
+                } else {
+                    [OneSignal onesignal_Log:ONE_S_LL_WARN message:@"appContext set and an old appId was found, attempting to call setAppId(oldAppId)"];
+                    [self setAppId:prevAppId];
+                }
+                return;
+            }
+        }
+    }
+
+    [OneSignal onesignal_Log:ONE_S_LL_VERBOSE message:@"setLaunchOptions(launchOptions) successful and appId is set, continuing OneSignal init..."];
```

Comment:
Let's say `initializing OneSignal` here instead of `continuing OneSignal init...`

Reply:
Got it!